### DISCUSSION TITLE
License question: LPGL v3.0 and Apple App Store

### DISCUSSION THREAD

**Discussion by sohels** [openpgpjs/openpgpjs#1208] [Q&A]
Hi, wondering if its ok to embed this library into an app that will be distributed via the Apple App Store? There appears to be some concern / confusion about the restrictions on distribution in the [App Store terms](https://www.apple.com/legal/internet-services/itunes/us/terms.html) and whether there is conflict with GPL / LGPL. To address this, Open Whisper Systems, for example, have [explicitly authorized](https://whispersystems.org/blog/license-update/) distributing apps via the App Store. Would be great if you could clarify this aspect.

**Comment by bartbutler** [comment]
Hi @sohels,

I certainly have no objection to using OpenPGP.js in applications distributed via the Apple App store, though I question why you'd want to use a JS library rather than a native C library as performance tends to be crucial for crypto applications. That said, some of the components of OpenPGP.js are LGPL-licensed by their respective authors and I do not think we can necessarily speak for them from a legal standpoint. That said, I can't imagine anyone chose LGPL with the intention of preventing it. Sorry if that is not particularly helpful :(
